x-pack/plugins/security_solution/server/fleet_integration/handlers/remove_policy_from_artifacts.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import pMap from 'p-map';
+
+import {
+  ENDPOINT_TRUSTED_APPS_LIST_ID,
+  ENDPOINT_EVENT_FILTERS_LIST_ID,
+  ENDPOINT_HOST_ISOLATION_EXCEPTIONS_LIST_ID,
+} from '@kbn/securitysolution-list-constants';
+import { ExceptionListClient } from '../../../../lists/server';
+import { PostPackagePolicyDeleteCallback } from '../../../../fleet/server';
+
+export const ARTIFACT_LISTS_IDS_TO_REMOVE = [
+  ENDPOINT_TRUSTED_APPS_LIST_ID,
+  ENDPOINT_EVENT_FILTERS_LIST_ID,
+  ENDPOINT_HOST_ISOLATION_EXCEPTIONS_LIST_ID,
+];
+
+/**
+ * Removes policy from artifacts
+ */
+export const removePolicyFromArtifacts = async (
+  exceptionsClient: ExceptionListClient,
+  policy: Parameters<PostPackagePolicyDeleteCallback>[0][0]
+) => {
+  let page = 1;
+
+  const findArtifactsByPolicy = (currentPage: number) => {
+    return exceptionsClient.findExceptionListsItem({
+      listId: ARTIFACT_LISTS_IDS_TO_REMOVE,
+      filter: ARTIFACT_LISTS_IDS_TO_REMOVE.map(
+        () => `exception-list-agnostic.attributes.tags:"policy:${policy.id}"`
+      ),
+      namespaceType: ARTIFACT_LISTS_IDS_TO_REMOVE.map(() => 'agnostic'),
+      page: currentPage,
+      perPage: 50,
```

Comment:
I would suggest increasing the this number - maybe to 1000?

Reply:
there could be - we just don't know. But the main reason I suggest a higher value to not do as many async calls to ES and keep the time it takes to process them down. I think we use 1000 in other places as well

Reply:
Got it, will change that number in my next PR, thanks for the advise! 🙂